src/services/offer.js

```diff
@@ -192,29 +196,69 @@ class OfferService {
+     *
      * @param {*} options
      * value: Text to use in full-text-search
-     * offset: Point to start looking (and limiting)
+     * queryToken: Id of the last fetched offer
      * limit: How many offers to show
      * jobType: Array of jobTypes allowed
      */
-    get({ value = "", offset = 0, limit = OfferService.MAX_OFFERS_PER_QUERY, showHidden = false, showAdminReason = false, ...filters }) {
-
-        const offers = (value ? Offer.find(
-            { "$and": [this._buildFilterQuery(filters), { "$text": { "$search": value } }] }, { score: { "$meta": "textScore" } }
-        ) : Offer.find(this._buildFilterQuery(filters))).current();
+    async get({ value = "", queryToken = null, limit = OfferService.MAX_OFFERS_PER_QUERY,
+        showHidden = false, showAdminReason = false, ...filters }) {
+
+        const {
+            id: lastOfferId,
+            score: lastOfferScore
+        } = queryToken ? this.decodeQueryToken(queryToken) : {};
+
+        let offers;
+        if (lastOfferId && value) {
+            offers = Offer.aggregate([
+                { $match: { $text: { $search: value } } },
+                { $addFields: {
+                    score: { $meta: "textScore" },
+                    adminReason: { $cond: [showAdminReason, "$adminReason", "$$REMOVE"] }
+                } },
+                { $match: { "$or": [
+                    { score: { "$lt": lastOfferScore } },
+                    { score: lastOfferScore, _id: { "$gt": ObjectId(lastOfferId) } }
+                ] } },
+                { $match: Offer.filterCurrent() },
+                { $match: showHidden ? {} : Offer.filterNonHidden() }
+            ]);
+
+        } else {
+            if (lastOfferId) {
+
+                offers = Offer.find({ "$and": [
+                    this._buildFilterQuery(filters),
+                    { _id: { "$gt": ObjectId(lastOfferId) } }
+                ] });
+
+            } else {
+
+                offers = (value ? Offer.find({ "$and": [
+                    this._buildFilterQuery(filters),
+                    { "$text": { "$search": value } }
+                ] }, { score: { "$meta": "textScore" } }
+
+                ) : Offer.find(this._buildFilterQuery(filters)));
+            }
 
-        if (!showHidden) offers.withoutHidden();
+            offers.current();
+            if (!showHidden) offers.withoutHidden();
+            if (!showAdminReason) offers.select("-adminReason");
+        }
 
-        const offersQuery = offers
-            .sort(value ? { score: { "$meta": "textScore" } } : undefined)
-            .skip(offset)
+        const results = await offers
+            .sort(value ? { score: { "$meta": "textScore" }, _id: 1 } : { _id: 1 })
             .limit(limit)
         ;
 
-        return showAdminReason ? offersQuery : offersQuery.select("-adminReason");
-
+        return results.map(this.buildQueryToken);
```

Comment:
no need to have the token per offer, simply return an object instead with two fields: the search `results` and the `queryToken`. Notify whoever is working on the frontend to update the way they receive the search results (accessing the results array, instead of the offers directly)

Reply:
@TiagooGomess Can you take care of this in https://github.com/NIAEFEUP/nijobs-fe/pull/172 ?

Reply:
Sure

Reply:
It's already done in https://github.com/NIAEFEUP/nijobs-fe/pull/172. This change facilitates my work by some orders of magnitude.